File: app/models/project.rb
class Project < ApplicationRecord
  has_many :editors, dependent: :destroy
  has_many :entities, dependent: :destroy
  has_many :languages, dependent: :destroy
  has_many :operating_systems, dependent: :destroy
  validates :date, presence: true, uniqueness: { scope: [:name] }
  validates :name, presence: true
  validates :total_seconds, presence: true
end

File: server/src/models/project.js
'use strict';
module.exports = function(sequelize, DataTypes) {
  var Project = sequelize.define('Project', {
      date: {
        allowNull: false,
        type: DataTypes.DATE
      },
      name: {
        allowNull: false,
        type: DataTypes.STRING
      },
      totalSeconds: {
        allowNull: false,
        type: DataTypes.INTEGER
      },
  }, {
    classMethods: {
      associate: function(models) {
          models.Project.hasMany(models.Editor, { foreignKey: 'projectId', sourceKey: 'id', as: 'editors' });
          models.Project.hasMany(models.Entity, { foreignKey: 'projectId', sourceKey: 'id' });
          models.Project.hasMany(models.Language, { foreignKey: 'projectId', sourceKey: 'id' });
          models.Project.hasMany(models.OperatingSystems, { foreignKey: 'projectId', sourceKey: 'id' });
      }
    }
  });
  return Project;
};